src/Microsoft.ML.Data/Transforms/ValueMapping.cs

```diff
@@ -396,29 +414,32 @@ internal sealed class Options
         }
 
         internal ValueMappingTransformer(IHostEnvironment env, IDataView lookupMap,
-            string keyColumn, string valueColumn, (string outputColumnName, string inputColumnName)[] columns)
+            DataViewSchema.Column lookupKeyColumn, DataViewSchema.Column lookupValueColumn, (string outputColumnName, string inputColumnName)[] columns)
             : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(ValueMappingTransformer)), columns)
         {
-            Host.CheckNonEmpty(keyColumn, nameof(keyColumn), "A key column must be specified when passing in an IDataView for the value mapping");
-            Host.CheckNonEmpty(valueColumn, nameof(valueColumn), "A value column must be specified when passing in an IDataView for the value mapping");
-            _valueMap = CreateValueMapFromDataView(lookupMap, keyColumn, valueColumn);
-            int valueColumnIdx = 0;
-            Host.Check(lookupMap.Schema.TryGetColumnIndex(valueColumn, out valueColumnIdx));
-            ValueColumnMetadata = lookupMap.Schema[valueColumnIdx].Annotations;
+            _lookupKeyColumn = lookupKeyColumn;
+            _lookupValueColumn = lookupValueColumn;
+            _valueMap = CreateValueMapFromDataView(lookupMap, _lookupKeyColumn, _lookupValueColumn);
+            ValueColumnMetadata = lookupMap.Schema[_lookupValueColumn.Index].Annotations;
 
```

Comment:
Does _lookupValueColumn.Annotations work?

Reply:
Just did a small refactorization around `Annotations` and extent a test to cover it.

---
In reply to: [271578470](https://github.com/dotnet/machinelearning/pull/3180#discussion_r271578470) [](ancestors = 271578470)